### PR TITLE
Trrack migrate

### LINKED ISSUE #111: De-duplicate and improve undo/redo
Remove the undo/redo from the menu on the left, and make it work in the tool bar. 

Also make the buttons stateful (grayed out if not available)

Provide a tool-tip

### LINKED ISSUE #151: Add "Show Element View" to meatball menu. Remove hamburger menu

### PR DESCRIPTION
**Pull request by JakeWags**
### Does this PR close any open issues?
Closes #111 
Closes #151

### Give a longer description of what this PR addresses and why it's needed
This PR migrates the trrack instance to `packages/app` and moves the undo/redo and prov vis buttons to the toolbar. Also, the PR includes the restyling of the header to match the multinet style.

### Provide pictures/videos of the behavior before and after these changes (optional)
New Style:
image

### Are there any additional TODOs before this PR is ready to go?

TODOs:
- Better logo image file loading

### CODE REVIEW

**Comment by JakeWags** [comment]
commit adb1db8 closes #151

File: packages/core/src/sort.ts [hunk 1]
```diff
@@ -18,9 +18,9 @@ function sortByCardinality(rows: Intersections) {
 function sortByDegree(rows: Intersections) {
   const { values, order } = rows;
   const newOrder = [...order].sort(
-    (a, b) =>
-      getDegreeFromSetMembership(values[a].setMembership) -
-      getDegreeFromSetMembership(values[b].setMembership),
+    (a, b) => {
+      return getDegreeFromSetMembership(values[a].setMembership) -
+      getDegreeFromSetMembership(values[b].setMembership)}
```
Comment (JackWilb) [inline packages/core/src/sort.ts:23]: This is unnecessary and might cause conflicts with our recent work

File: packages/app/src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,50 @@
...
+
+  const conf = useMemo(() => {
+    if (data !== null) {
+      const conf = {...config}
+      if (config.visibleSets.length === 0) {
+        const setList = Object.keys(data.sets);
+        conf.visibleSets = setList.slice(0, 6);
+      }
+
+      conf.visibleAttributes = data.attributeColumns.slice(0, 3);
```
Comment (JackWilb) [inline packages/app/src/App.tsx:33]: Does this overwrite the imported state?

File: packages/app/src/components/Root.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+export const ProvenanceContext = createContext<{
+    provenance: UpsetProvenance;
+    actions: UpsetActions;
+    isAtLatest: boolean;
+    isAtRoot: boolean;
+  }>(undefined!);
+
+export const Root = ({provenance, actions, data, config}: Props) => {
+    const ref = useRef<HTMLDivElement>(null);
```
Comment (JackWilb) [inline packages/app/src/components/Root.tsx:31]: This is a bad variable name — it's not descriptive enough. Can we update this to make it a bit more clear?

File: packages/app/src/components/Header/index.tsx [hunk 4]
```diff
@@ -1,43 +1,106 @@
...
+          >
+            Load Data
+          </Button>
+          <MoreVertIcon onClick={(e) => handleMenuClick(e)}></MoreVertIcon>
+          <Menu open={isMenuOpen} onClose={handleMenuClose} anchorEl={anchorEl}>
+            <MenuItem onClick={() => setShowImportModal(true) } color="inherit">
+                Import State
+            </MenuItem>
+            <MenuItem onClick={() => exportStateGrammar(provenance)} color="inherit">
+                Export
```
[the remark this answers is not in the document]
Comment (JakeWags) [reply packages/app/src/components/Header/index.tsx:76]: The extended options are in #145. Once this is merged, that PR needs to be updated to account for the migration.

**Comment by JakeWags** [comment]
> Another more general question, what happens to the provenance when you import a new config? Does it fully reset?

No. The state for importing is: "Replace State". The history is maintained and can still be traversed.

image

image

File: packages/app/src/App.tsx (hunk 2 above)
Comment (JakeWags) [reply packages/app/src/App.tsx:33]: No. See this state imported with only one visible set:
image

Comment (JackWilb) [reply packages/app/src/App.tsx:33]: Visible attributes is the `ReleaseDate, etc.` columns, right?

Comment (JakeWags) [reply packages/app/src/App.tsx:33]: This does not overwrite the attributes after importing.
I exported a state that had NO attributes visible.
Before importing the aforementioned state: 
image

After importing the state:
image

**Review by JackWilb — APPROVED** [review]

**Pull request merged**